client/components/forms/form-text-input/index.jsx

```diff
@@ -17,24 +14,32 @@ export default class FormTextInput extends PureComponent {
 		className: PropTypes.string,
 	};
 
-	constructor() {
-		super( ...arguments );
+	currentTextField = undefined;
 
-		this.selectOnFocus = this.selectOnFocus.bind( this );
-	}
+	textFieldRef = element => {
+		if ( element && this.props.inputRef ) {
```

Comment:
Don't check the `element` here: unsetting the ref to `null` when the referenced element is unmounted is a part of the game 🙂

Reply:
You're right, thanks!

The more I fix on this, the odder it seems to me. It isn't a very trivial thing to implement, but it seems like something you could need quite often; that is, a way of providing a ref to the outside while making use of it inside the component too. Is there a more idiomatic way of doing this?

Reply:
Yes, I'm also suprised that setting an external ref and using it internally at the same time is so complicated and that I have to re-implement a non-trivial part of React code.